compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -860,6 +844,60 @@ private BLangStatementExpression createIfElseFromConfigurable(BLangSimpleVariabl
     }
 
+    private List<BLangVariable> desugarGlobalVariables(List<BLangVariable> globalVars,
+                                                       BLangBlockFunctionBody initFnBody) {
+        List<BLangVariable> desugaredGlobalVarList = new ArrayList<>();
+
+        globalVars.forEach(globalVar -> {
+            // This will convert complex variables to simple variables.
+            switch (globalVar.getKind()) {
+                case TUPLE_VARIABLE:
+                    BLangNode blockStatementNode = rewrite(globalVar, env);
+                    for (BLangStatement bLangStatement : ((BLangBlockStmt) blockStatementNode).stmts) {
+                        if (bLangStatement.getKind() == NodeKind.FOREACH) {
+                            initFnBody.stmts.add(rewrite(bLangStatement, this.initFunctionEnv));
+                            continue;
+                        }
+                        BLangSimpleVariableDef simpleVarDef = (BLangSimpleVariableDef) bLangStatement;
+                        simpleVarDef.var.annAttachments = globalVar.getAnnotationAttachments();
+                        rewrite(simpleVarDef, env);
+                        addToInitFunction(simpleVarDef.var, initFnBody);
+                        desugaredGlobalVarList.add(simpleVarDef.var);
+                    }
+                    break;
+                default:
+                    long globalVarFlags = globalVar.symbol.flags;
+                    BLangSimpleVariable simpleGlobalVar = (BLangSimpleVariable) globalVar;
+                    if (Symbols.isFlagOn(globalVarFlags, Flags.CONFIGURABLE)) {
+                        if (Symbols.isFlagOn(globalVarFlags, Flags.REQUIRED)) {
+                            // If it is required configuration get directly
+                            List<BLangExpression> args = getConfigurableLangLibInvocationParam(simpleGlobalVar);
+                            BLangInvocation getValueInvocation = createLangLibInvocationNode("getConfigurableValue",
+                                    args, symTable.anydataType, simpleGlobalVar.pos);
+                            simpleGlobalVar.expr = getValueInvocation;
+                        } else {
+                            // If it is optional configuration create if else
+                            simpleGlobalVar.expr = createIfElseFromConfigurable(simpleGlobalVar);
+                        }
+                    }
+                    addToInitFunction(simpleGlobalVar, initFnBody);
+                    desugaredGlobalVarList.add(simpleGlobalVar);
+                    break;
+            }
+        });
+
+        return desugaredGlobalVarList;
+    }
+
+    private void addToInitFunction(BLangSimpleVariable globalVar, BLangBlockFunctionBody initFnBody) {
+        if (globalVar.expr == null) {
+            return;
+        }
+        BLangAssignment assignment = createAssignmentStmt(globalVar);
+        initFnBody.stmts.add(assignment);
+        globalVar.expr = null;
```

Comment:
Why do we need to make the expr null here?

Reply:
Global variables are initialized inside the init method so remove the expression and add it to global variables. This concept was there before this PR I have just moved it here.